commands/apps.go

```diff
@@ -393,6 +403,69 @@ func RunAppsDelete(c *CmdConfig) error {
 	return nil
 }
 
+// RunAppsDetect detects an function project and converts it into apps project.
+func RunAppsDetect(c *CmdConfig) error {
+	source, err := c.Doit.GetString(c.NS, doctl.ArgProjectSource)
+	if err != nil {
+		return err
+	}
+	if len(source) == 0 {
+		return fmt.Errorf("source cannot be empty")
+	}
+
+	sha, err := c.Doit.GetString(c.NS, doctl.ArgCommitHash)
+	if err != nil {
+		return err
+	}
+
+	name, err := c.Doit.GetString(c.NS, doctl.ArgProjectName)
+	if err != nil {
+		return err
+	}
+	if len(name) == 0 {
+		return fmt.Errorf("name cannot be empty")
+	}
+
+	branch, err := c.Doit.GetString(c.NS, doctl.ArgProjectBrach)
+	if err != nil {
+		return err
+	}
+	if len(branch) == 0 {
+		return fmt.Errorf("branch cannot be empty")
+	}
+
+	// Need to check, How user value will be overrided
+	autoDeploy, err := c.Doit.GetBool(c.NS, doctl.ArgDeployOnPush)
+	if err != nil {
+		return err
+	}
+	if len(c.Args) > 0 {
+		fmt.Println(c.Args[0])
+		x, err := strconv.ParseBool(c.Args[0])
+		if err == nil {
+			autoDeploy = x
+		} else {
+			return fmt.Errorf("expected true/false for deployonpush, received : %s", c.Args[0])
+		}
+	}
+
+	spec, err := c.Apps().Detect(source, sha, name, branch, autoDeploy)
+	if err != nil {
+		return err
+	}
+
+	switch Output {
+	case "json":
+		e := json.NewEncoder(c.Out)
+		e.SetIndent("", "  ")
+		return e.Encode(spec)
+	case "text":
+		return nil
```

Comment:
(assuming this is just not implemented yet) just a reminder to either error out if not supported or implemented output support